src/CowOrder.py

```diff
@@ -0,0 +1,92 @@
+from eip712_structs import EIP712Struct, Address, Boolean, Bytes, String, Uint
+import requests
+import json
+from eip712_structs import make_domain
+from web3.auto import w3
+from eth_account.messages import encode_defunct
+from hexbytes import HexBytes
+import time
+
+class Fee: 
+    def __init__(self, base_url, feeAddress):
+        self.base_url = base_url
+        self.feeAddress = feeAddress
+    
+    def setFee(self):
+        requestFee = requests.get(base_url + feeAddress + sellToken + "&buyToken=" + buyToken + "&amount=" +sellAmount + "&kind=" + kind)
+        feeAmount = int(requestFee.json()["amount"])
+        return feeAmount
+
+class Signature:
+    def __init__(self, privateKey, name, version, chainId, verifyingContract):
+        self.privateKey = privateKey
+        self.name = name
+        self.version = version
+        self.chainId = chainId
+        self.verifyingContract = verifyingContract     
+    
+    def getSig(self):
+        my_bytes = self.signable_bytes(self.getDomain(name, version, chainId, verifyingContract))
+        hash = w3.keccak(my_bytes)
+        message = encode_defunct(primitive=hash)
+        signed_message = w3.eth.account.sign_message(message, private_key=privateKey)
+        return signed_message.signature.hex()
+
+    def getDomain(self, name, version, chainId, verifyingContract):
+        return make_domain(name=name, version=version, chainId=chainId, verifyingContract=verifyingContract)
+
+class Order(Fee, Signature, EIP712Struct):
+    def __init__(self, sellToken, buyToken, sellAmount, kind, receiver, buyAmount, validTo, appData, partiallyFillable, sellTokenBalance, buyTokenBalance, signingScheme, orders):
+        self.sellToken: Address() = sellToken
+        self.buyToken: Address() = buyToken
+        self.sellAmount: Uint(256) = sellAmount
+        self.kind: String() = kind
+        self.receiver: Address() = receiver
+        self.buyAmount: Uint(256) = buyAmount
+        self.validTo: Uint(32) = validTo
+        self.appData: Bytes(32) = appData
+        self.feeAmount: Uint(256) = str(Fee.setFee(self))
+        self.partiallyFillable: Boolean() = partiallyFillable
+        self.signature = Signature.getSig(self)
+        self.signingScheme = signingScheme
+        self.sellTokenBalance: String() = sellTokenBalance
+        self.buyTokenBalance: String() = buyTokenBalance 
+    
+    def toJson(self):
+        jsonorder = json.dumps(self, default=lambda obj: obj.__dict__, indent=4)
+        self.orders = orders
+        self.base_url = base_url
+        return requests.post(self.base_url + self.orders + jsonorder)
```

Comment:
If we look at the post made [here]() in the original script 

```
r = requests.post(base_url + "orders", json=j)
```
this is a string concatenation of the base url with the string "orders" followed by the json data of the order

```suggestion
        return requests.post("{base_url}orders".format(base_url), json=jsonorder)
```